src/builtin/function/mod.rs

```diff
@@ -11,4 +11,27 @@ pub(crate) use self::uuidv4::uuidv4;
 mod now;
 mod uuidv4;
 
-pub type FunctionFn = fn(&HashMap<String, Value>, &Context) -> Result<Value>;
+/// Evaluation engine function signature
+///
+/// You can register custom function with the [`function`] method.
+///
+/// # Arguments
+///
+/// * `args` - A function arguments
+/// * `context` - An evaluation context
+///
+/// # Examples
```

Comment:
This may benefit from a longer code example, e.g. usage with `function` maybe ?

Reply:
Links added to `function` / `filter` methods where one can find full example how to register `function` / `filter`.